test/e2e/storage-endpoint.integration.test.js

```diff
@@ -0,0 +1,152 @@
+        expect(res.body).toEqual(thatIsUploadResponse);
+      });
+  }, 5000);
+
+  it("should create document when placeholder object is there", async () => {
+    await request
+      .get("/storage/create")
+      .set("Content-Type", "application/json")
+      .set("Accept", "application/json")
+      .send({
+        document: ropstenDocument,
+        id: "123"
+      })
+      .expect("Content-Type", /json/)
+      .expect(200)
+      .expect(async res => {
+        await remove({ Bucket: config.bucketName, Key: "123" });
+        expect(res.body.id).toEqual("123");
+      });
+  }, 5000);
+
+  it("should retrieve the document created", async () => {
+    try {
+      await request
+        .get("/storage/create")
+        .set("Content-Type", "application/json")
+        .set("Accept", "application/json")
+        .send({
+          document: ropstenDocument,
+          id: "123"
+        })
+        .expect("Content-Type", /json/)
+        .expect(200);
+
+      await request
+        .get("/storage/get/123")
+        .set("Content-Type", "application/json")
+        .expect("Content-Type", /json/)
+        .expect(200)
+        .expect(async res => {
+          await remove({ Bucket: config.bucketName, Key: "123" });
+          expect(res.body).toEqual(thatIsRetrievedDocument);
+        });
+    } catch (e) {
+      await remove({ Bucket: config.bucketName, Key: "123" });
+      throw e;
+    }
+  }, 5000);
+
+  it("should throw error forbidden when directly access the document", async () => {
+    const document = { foo: "bar" };
+    const params = {
+      Bucket: config.bucketName,
+      Key: "123",
+      Body: JSON.stringify({ document })
+    };
+
+    try {
+      const uploaded = await put(params);
+      await supertest(`${uploaded.Location}`)
+        .get("/")
+        .set("Content-Type", "application/json")
+        .expect("Content-Type", /json/)
+        .expect(403);
+
+      await remove(params);
+    } catch (e) {
+      await remove(params);
+      throw e;
+    }
+  }, 5000);
+
+  it("should create a placeholder object", async () => {
+    let queueId = "";
+    try {
+      await request
+        .get("/storage/queue")
+        .set("Content-Type", "application/json")
+        .expect("Content-Type", /json/)
+        .expect(200)
+        .expect(res => {
+          queueId = res.body.queueNumber;
+          expect(res.body).toEqual({
+            queueNumber: expect.any(String),
+            key: expect.any(String)
+          });
+        });
+
+      await request
+        .get(`/storage/get/${queueId}`)
+        .set("Content-Type", "application/json")
+        .expect(400);
+
+      await supertest(`https://${config.bucketName}.s3.amazonaws.com`)
+        .get(`/${queueId}`)
+        .set("Content-Type", "application/json")
+        .expect(403);
+
+      const params = {
+        Bucket: config.bucketName,
+        Key: queueId
+      };
+
+      await remove(params);
```

Comment:
use after each instead

Reply:
I dont know the id of the document create in some of test cases. Cant remove document there.

Reply:
set the id into a hoisted variable

```
  let variable
  afterEach(() => clean(variable))
  test(() => variable = 12)
```

Reply:
refactored.